+Base/QRconv.m

```diff
@@ -0,0 +1,384 @@
+    fil =  B*circleFunc(XX, YY, 0,       0,      r) ...
+        - sn*circleFunc(XX, YY, 1*lx/8,  1*ly/8, r/3) ...
+        + sp*circleFunc(XX, YY, lx/4,    ly/4,   r/3) ...
+        - sn*circleFunc(XX, YY, 3*lx/8,  3*ly/8, r/3) ...
+        + sp*circleFunc(XX, YY, lx/2,    ly/2,   r/3) ...
+        - sn*circleFunc(XX, YY, 5*lx/8,  5*ly/8, r/3) ...
+        + sp*circleFunc(XX, YY, 3*lx/4,  3*ly/4, r/3) ...
+        - sn*circleFunc(XX, YY, 7*lx/8,  7*ly/8, r/3) ...
+        + B*circleFunc(XX, YY, lx,      ly,     r);
+
+    % Normalize
+    S1 = size(fil);
+    fil = fil - sum(sum(fil))/S1(1)/S1(2);
+    fil = fil / sqrt(sum(sum(fil.^2)));
+
+    % Uncomment this line to get a better idea of what the filter looks like.
+%     imwrite(.5 + fil/max(max(fil))/2, 'fil.png');
+
+    lx = ceil(l*ca);
+    ly = ceil(l*sa);
+
+    % The heart of the function, doing the convolution on the images with our filter, on both arms (H&V) of the QRs.
+    convH = conv2(img, fil);
+    convV = conv2(img, rot90(fil));
+
+    S = size(img);
+    X = (1:S(1)) + r;
+    Y = (1:S(2)) + r;
+
+    convH = convH(X + invx*ly, Y + invy*lx);
+    convV = convV(X + (~invy)*lx, Y + invx*ly);
+
+    conv = convH.*convH.*convH + convV.*convV.*convV;
+end
+function [cx, cy, CX, CY] = findQRs(bw, conv, ang0, r, l, V_expected)
+    S = size(conv);
+
+    ca0 = cos(ang0);
+    sa0 = sin(ang0);
+
+    lxx = l*(sa0+ca0)/2;
+    lyy = l*(sa0-ca0)/2;
+    
+    [XX, YY] = meshgrid(1:S(2), 1:S(1));
+
+    CC = bwconncomp(conv > max(max(conv))/8);
+
+    NQR = CC.NumObjects;
+    cx = NaN(1,NQR);
+    cy = NaN(1,NQR);
+
+    for ii = 1:NQR
+        cx(ii) = mean(XX(CC.PixelIdxList{ii}));
+        cy(ii) = mean(YY(CC.PixelIdxList{ii}));
+    end
+    
+    pad = abs(l*(sa0+ca0)/2) + 2*r;
+
+    isQR = true(1, NQR);
+    
+    bitcoord = .75*(-2:2)/6.25;
+    % Replace with meshgrid?
+    bity = repmat(-bitcoord,  [5 1]);
+    bitx = repmat(-bitcoord', [1 5]);
+
+    A = l * [ca0, sa0; -sa0, ca0];
+
+    BITX = A(1,1) * bitx + A(2,1) * bity;
+    BITY = A(1,2) * bitx + A(2,2) * bity;
+
+    CX = NaN*cx;
+    CY = NaN*cx;
+    
+    m_vectors = NaN(25, NQR);
+
+    for ii = 1:NQR
+        if cx(ii) + lxx < pad || cx(ii) + lxx > S(2) - pad || cy(ii) + lyy < pad || cy(ii) + lyy > S(1) - pad
+            isQR(ii) = false;   % QR is clipping the edge of screen and decoding should not be attempted.
+        else
+            m = NaN(5);
+            bitave = -1:1;
+
+            for jj = 1:length(bitcoord)         % TODO: Replace for loop with one-liner.
+                for kk = 1:length(bitcoord)
+                    m(kk,jj) = mean(mean(bw(round(BITY(jj,kk) + cy(ii) + lyy) + bitave, round(BITX(jj,kk) + cx(ii) + lxx) + bitave))) > .5;
+                end
+            end
+
+            [CX(ii), CY(ii), ~, isQR(ii)] = interpretQR(m(:));
+            
+            if ~isQR(ii) && ~any(isnan(V_expected))
+                dist = 2;
+                
+                for jj = 1:25
```

Comment:
do we care about version controlling with QR generation code? that's a huge step i recognize, but a lot of these params are hard coded

Reply:
It can be a future enhancement.